python-backend/app/api/mines/tailings/resources/tailings.py

```diff
@@ -12,56 +12,46 @@
 from app.api.utils.url import get_documents_svc_url
 
 from ..models.tailings import MineTailingsStorageFacility
-from ....documents.namespace.documents import api as doc_api
+from app.api.mines.mine.models.mine import Mine
 
+from app.api.mines.mine_api_models import MINE_TSF_MODEL
 
-class MineTailingsStorageFacilityResource(Resource, UserMixin, ErrorMixin):
+
+class MineTailingsStorageFacilityListResource(Resource, UserMixin):
     parser = reqparse.RequestParser()
-    parser.add_argument('mine_guid', type=str, help='mine to create a new tsf on')
     parser.add_argument(
-        'tsf_name',
+        'mine_tailings_storage_facility_name',
         type=str,
         trim=True,
         help='Name of the tailings storage facility.',
-        store_missing=False)
+        required=True)
 
-    @api.doc(
-        params={
-            'mine_tailings_storage_facility_guid':
-            'mine_tailings_storage_facility_guid to be retrieved, or return error if not provided'
-        })
+    @api.doc(description='Gets the tailing storage facilites for the given mine')
+    @api.marshal_with(
+        MINE_TSF_MODEL, envelope='mine_tailings_storage_facilities', as_list=True, code=200)
     @requires_role_mine_view
-    def get(self, mine_tailings_storage_facility_guid=None):
-        if mine_tailings_storage_facility_guid:
-            tsf = MineTailingsStorageFacility.find_by_tsf_guid(mine_tailings_storage_facility_guid)
-            if not tsf:
-                return self.create_error_payload(404, 'mine_tailings_storage_facility not found')
-            return tsf.json()
-        else:
-            mine_guid = request.args.get('mine_guid', type=str)
-            mine_tsf_list = MineTailingsStorageFacility.find_by_mine_guid(mine_guid)
-            if mine_tsf_list is None:
-                return self.raise_error(404, 'Mine_guid or tsf_guid must be provided')
-            return {
-                'mine_storage_tailings_facilities': list(map(lambda x: x.json(), mine_tsf_list))
-            }
+    def get(self, mine_guid):
+        mine = Mine.find_by_mine_guid(mine_guid)
+        if not mine:
+            raise NotFound('mine not found')
```

Comment:
Trivial, but we should probably be consistent with our punctuation in response messages. Most errors are capitalizing the first letter and using periods at the end of sentences.